crates/nu-command/src/bytes/at.rs

```diff
@@ -159,3 +156,116 @@ fn action(input: &Value, args: &Arguments, head: Span) -> Value {
         ),
     }
 }
+
+fn handle_byte_stream(
+    args: &Arguments,
+    stream: ByteStream,
+    call: &Call,
+    metadata: Option<nu_protocol::PipelineMetadata>,
+    engine_state: &EngineState,
+) -> Result<PipelineData, ShellError> {
+    let idxs = args.indexes;
+    match stream.reader() {
+        Some(reader) => {
+            let iter = reader.bytes();
+
+            if idxs.0 < 0 || idxs.1 < 0 {
+                match iter.try_len() {
+                    Ok(_) => {
+                        let vec = iter.filter_map(Result::ok).collect::<Vec<u8>>();
+                        Ok(read_bytes(&vec, &idxs, call.head).into_pipeline_data_with_metadata(metadata))
+                    }
+                    _ => Err(ShellError::IncorrectValue {
+                        msg:
+                            "Negative range values cannot be used with streams that don't specify a length"
+                                .into(),
+                        val_span: call.head,
+                        call_span: call.arguments_span(),
+                    }),
+                }
+            } else {
+                Ok(read_stream(iter, idxs, call, engine_state, metadata))
+            }
+        }
+        None => Ok(PipelineData::empty()),
+    }
+}
+
+fn read_bytes(val: &[u8], range: &Subbytes, head: Span) -> Value {
+    let len = val.len() as isize;
+    let start = if range.0 < 0 { range.0 + len } else { range.0 };
+    let end = if range.1 < 0 { range.1 + len } else { range.1 };
+
+    if start > end {
+        Value::binary(vec![], head)
+    } else {
+        let val_iter = val.iter().skip(start as usize);
+        Value::binary(
+            if end == isize::MAX {
+                val_iter.copied().collect::<Vec<u8>>()
+            } else {
+                val_iter.take((end - start + 1) as usize).copied().collect()
+            },
+            head,
+        )
+    }
+}
+
+fn read_stream(
+    iter: Bytes<Reader>,
+    range: Subbytes,
+    call: &Call,
+    engine_state: &EngineState,
+    metadata: Option<nu_protocol::PipelineMetadata>,
+) -> PipelineData {
+    let start = range.0 as usize;
+    let end = (range.1 - range.0) as usize;
+    let mut iter = iter.skip(start).take(end);
+
+    let stream = ByteStream::from_fn(
+        call.head,
+        engine_state.signals().clone(),
+        ByteStreamType::Binary,
+        move |buf| match iter.next() {
+            Some(Ok(n)) if n > 0 => match buf.write(&[n]) {
+                Ok(_) => Ok(true),
+                Err(err) => Err(err.into()),
+            },
+            _ => Ok(false),
+        },
+    );
+
+    PipelineData::ByteStream(stream, metadata)
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Normally we put `nu!` relative tests into `nu-command/tests`.  For example, this can be put under `nu-command/tests/commands/bytes/at.rs`

Reply:
I have put them in `tests/commands/bytes/at.rs`